### PR TITLE
Enhance internal logging

### LINKED ISSUE #3866: Add `Warn` internal logging function
> Ended up not creating a `Warn` function as mentioned in the issue since Info is already being used for warning messages in other places.

_Originally from https://github.com/open-telemetry/opentelemetry-go/issues/3854_

We should consider adding `Warn` logging function to `go.opentelemetry.io/otel/internal/global`.

### PR DESCRIPTION
**Pull request by unicod3**
Resolves #3866 

I've used the table below as suggested on the issue:

Name | LogR Severity
-- | -- 
Error | 0 
Warn | 1 
Info | 4 
Debug | 8

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -13,6 +13,7 @@ This project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.htm
 - The `WithoutTimestamps` option to `go.opentelemetry.io/otel/exporters/stdout/stdoutmetric` to sets all timestamps to zero. (#3828)
 - The new `Exemplar` type is added to `go.opentelemetry.io/otel/sdk/metric/metricdata`.
   Both the `DataPoint` and `HistogramDataPoint` types from that package have a new field of `Exemplars` containing the sampled exemplars for their timeseries. (#3849)
+- The New `Warn` function is added to `internal/global` package for internal logging. (#3900)
```
Comment (MrAlias) [inline CHANGELOG.md:16]: This is probably not needed, the `internal` package is not accessible to the users we intend as the audience of this document. We should mention that the logging levels of the project have changed.

**Review by MrAlias — APPROVED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (unicod3) [reply CHANGELOG.md:16]: Would it be more appropriate if I changed it to something like: 


- The logging levels of the project are updated as below: (#3900) 
  | Name | Severity |
  | -- | -- |
  | Error | 0 |
  | Warn | 1 |
  | Info | 4 |
  | Debug | 8 |

Comment (pellared) [reply CHANGELOG.md:16]:
```suggestion
- The internal logging introduces a warning level equal to `V(8)` (verbosity level) . (#3900)
```

**Review by pellared — APPROVED** [review]

File: CHANGELOG.md [hunk 2]
```diff
@@ -13,6 +13,7 @@ This project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.htm
 - The `WithoutTimestamps` option to `go.opentelemetry.io/otel/exporters/stdout/stdoutmetric` to sets all timestamps to zero. (#3828)
 - The new `Exemplar` type is added to `go.opentelemetry.io/otel/sdk/metric/metricdata`.
   Both the `DataPoint` and `HistogramDataPoint` types from that package have a new field of `Exemplars` containing the sampled exemplars for their timeseries. (#3849)
+- The internal logging introduces a warning level equal to `V(8)` (verbosity level) . (#3900)
```
Comment (pellared) [inline CHANGELOG.md:16]:
```suggestion
- The internal logging introduces a warning level equal to `V(8)` (verbosity level). (#3900)
```

There was a typo in my suggestion. My bad, sorry.

**Comment by MadVikingGod** [comment]
I don't object to this, it's as good as any other, but where are we going to use the warn vs info?

File: CHANGELOG.md [hunk 3]
```diff
@@ -13,6 +13,7 @@ This project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.htm
 - The `WithoutTimestamps` option to `go.opentelemetry.io/otel/exporters/stdout/stdoutmetric` to sets all timestamps to zero. (#3828)
 - The new `Exemplar` type is added to `go.opentelemetry.io/otel/sdk/metric/metricdata`.
   Both the `DataPoint` and `HistogramDataPoint` types from that package have a new field of `Exemplars` containing the sampled exemplars for their timeseries. (#3849)
+- The internal logging introduces a warning level equal to `V(8)` (verbosity level). (#3900)
```
Comment (MadVikingGod) [inline CHANGELOG.md:16]:
```suggestion
- The internal logging introduces a warning level equal to `V(1)` (verbosity level). (#3900)
```

File: CHANGELOG.md
Comment (MadVikingGod) [inline CHANGELOG.md:1]: There should be a comment that info has changed to v4 and error to v8

File: CHANGELOG.md [hunk 4]
```diff
@@ -13,6 +13,7 @@ This project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.htm
 - The `WithoutTimestamps` option to `go.opentelemetry.io/otel/exporters/stdout/stdoutmetric` to sets all timestamps to zero. (#3828)
 - The new `Exemplar` type is added to `go.opentelemetry.io/otel/sdk/metric/metricdata`.
   Both the `DataPoint` and `HistogramDataPoint` types from that package have a new field of `Exemplars` containing the sampled exemplars for their timeseries. (#3849)
+- The internal logging changes the verbosity level of info to `V(4)`, the verbosity level of debug to `V(8)`, and introduces a warning level equal to `V(1)`. (#3900)
```
Comment (pellared) [inline CHANGELOG.md:16]: I suggest moving it to `Changed` section (instead of `Added`)

Comment (unicod3) [reply CHANGELOG.md]: yeap agreed, actually, I've split it into two parts now, one line for _changed_ one line for _added_ section

**Review by pellared — APPROVED** [review]

**Pull request merged**